sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1584,6 +1584,142 @@ -(void) testPullWithFetchLimitNotMultipleOfFifty
                   @"Invalid URL: %@", thirdRequest.URL.absoluteString);
 }
 
+-(void) testPullTable_UsePullSettings_NegativePageSize
+{
+    MSPullSettings *pullSettings = [MSPullSettings new];
+    pullSettings.pageSize = -1;
+    
+    NSString *pullRequest = [self getPullTableRequestWithPullSettings:pullSettings usePullSettings:YES];
+
+    XCTAssertEqualObjects(pullRequest, @"https://someUrl/tables/TodoNoVersion?$top=50&__includeDeleted=true&$skip=0&__systemProperties=__deleted", "Incorrect pull request");
+}
+
+-(void) testPullTable_UsePullSettings_ZeroPageSize
+{
+    MSPullSettings *pullSettings = [MSPullSettings new];
+    pullSettings.pageSize = 0;
+    
+    NSString *pullRequest = [self getPullTableRequestWithPullSettings:pullSettings usePullSettings:YES];
+    
+    XCTAssertEqualObjects(pullRequest, @"https://someUrl/tables/TodoNoVersion?$top=50&__includeDeleted=true&$skip=0&__systemProperties=__deleted", "Incorrect pull request");
```

Comment:
Due to the way we build URLs, the NSDictionary ordering can be different between the 2, so the URL may be ?A&B on 32 while be B&A on 64bit devices.  Usually this impacts any query that is using system properties iirc.